### PR TITLE
Add integration tests for add-ons data flow

### PR DESCRIPTION
**Pull request by wzieba**
### Description

This PR adds integration tests for the Add-ons data flow. I call it an "integration" as it covers `remote`->`domain`<->`database` flow.

See #2107 for subjects of tests.

### CODE REVIEW

**Review by ThomazFB — APPROVED** [review]
Tests are looking good and all green. LGTM :shipit:

File: plugins/woocommerce/src/test/java/org/wordpress/android/fluxc/store/ProvideAddonsIntegrationTests.kt [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+
+            sut.fetchAllGlobalAddonsGroups(siteModel)
+
+            assertThat(sut.observeAddonsForProduct(siteModel.siteId, product).first()).isEmpty()
+            verify(logger).e(API, "Can't map sample. CustomText Add-on has to have restrictions defined.")
+        }
+    }
+
+    @Test
+    fun name() {
```
Comment (shiki) [inline plugins/woocommerce/src/test/java/org/wordpress/android/fluxc/store/ProvideAddonsIntegrationTests.kt:524]: 😝

Comment (wzieba) [reply plugins/woocommerce/src/test/java/org/wordpress/android/fluxc/store/ProvideAddonsIntegrationTests.kt:524]: 😬 😬 good eye!

- [x] Fix test name

Comment (wzieba) [reply plugins/woocommerce/src/test/java/org/wordpress/android/fluxc/store/ProvideAddonsIntegrationTests.kt:524]: I just recalled: @ThomazFB noticed it in one of the later PRs 😄 https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2113

**Pull request merged**